ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/MaskKeepLeftKudf.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.ksql.function.udf.string;
+
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+
+@UdfDescription(name = "mask_keep_left", author = "Confluent",
+    description = "Returns a version of the input string with all but the"
+        + " specified number of left-most characters masked out."
+        + " Default masking rules will replace all upper-case characters with 'X', all lower-case"
+        + " characters with 'x', all digits with 'n', and any other character with '-'.")
+public class MaskKeepLeftKudf {
+
+  @Udf(description = "Returns a masked version of the input string. All characters except for the"
+      + " first n will be replaced according to the default masking rules.")
+  public String mask(final String input, final int numChars) {
+    return doMask(new Masker(), input, numChars);
+  }
+
+  @Udf(description = "Returns a masked version of the input string. All characters except for the"
+      + " first n will be replaced with the specified masking characters: e.g."
+      + " mask_keep_left(input, numberToKeep, upperCaseMask, lowerCaseMask, digitMask, otherMask)"
+      + " . Pass NULL for any of the mask characters to prevent masking of that character type.")
+  public String mask(final String input, final int numChars, final String upper, final String lower,
+      final String digit, final String other) {
+    // TODO once KSQL gains Char sql-datatype support we should change the xxMask params to int
+    // (codepoint) instead of String
+
+    // TODO really need a way for UDFs to do one-shot init() stuff instead of repeating all this
+    // literal-param manipulation and validation for every single record
+    final int upperMask = upper == null ? Masker.NO_MASK : upper.codePointAt(0);
```

Comment:
As this is repeated in the other classes, perhaps create a static method on `Masker` to create it rather than having to duplicate the code?

Reply:
good idea, thanks. done.